class.jetpack.php

```diff
@@ -537,7 +537,7 @@ private function __construct() {
 		 * Prepare Gutenberg Editor functionality
 		 */
 		require_once JETPACK__PLUGIN_DIR . 'class.jetpack-gutenberg.php';
-		add_action( 'init', array( 'Jetpack_Gutenberg', 'load_blocks' ) ); // Registers all the Jetpack blocks .
+		add_action( 'wp_loaded', array( 'Jetpack_Gutenberg', 'load_blocks' ) ); // Registers all the Jetpack blocks .
```

Comment:
Noting that sync related hooks might run on `init` so by doing this we might make loading blocks invisible for sync.

I also tested something like this but no dice:
```php
// `Jetpack_Gutenberg::load_blocks` registers all the Jetpack blocks.
// Priority 11 is to ensure all modules load first and register their block assets.
add_action( 'init', array( 'Jetpack_Gutenberg', 'load_blocks', 10000 ) );
```

Reply:
`wp_loaded` is a bit late for that IMHO. I think registration of blocks should happen early enough, this is how it's also done in Gutenberg: https://github.com/WordPress/gutenberg/blob/6e5dc2a9beb159351b5101fff562c5f81306f1b2/packages/block-library/src/categories/index.php#L102

If we want to delay the assets enqueueing (or anything else) though, we could separate it and hook it on the later hook specifically.

Reply:
>I also tested something like this but no dice: add_action( 'init', array( 'Jetpack_Gutenberg', 'load_blocks', 10000 ) );

`add_action( 'init', array( 'Jetpack_Gutenberg', 'load_blocks', 11 ) );`

worked for me 😕

Reply:
Turns out I had `,` in a wrong place in my priority change `add_action( 'init', array( 'Jetpack_Gutenberg', 'load_blocks', 10000 ) );` — thanks @enejb for spotting and shame on my sloppy coding. 😅 

With that change, loading assets works great: https://github.com/Automattic/jetpack/pull/10728/commits/382e9c8ec1748e243e26bd805bcdb1224e4a665d

Reply:
I've attempted to update the description to reflect this as it was out of date, but any details that can be added there would be appreciated.